Tests/Editor/Tracking/Collision/List/ReverseCollisionsOrderTest.cs

```diff
@@ -0,0 +1,86 @@
+﻿using VRTK.Core.Tracking.Collision;
+using VRTK.Core.Tracking.Collision.List;
+
+namespace Test.VRTK.Core.Tracking.Collision.List
+{
+    using UnityEngine;
+    using NUnit.Framework;
+    using Test.VRTK.Core.Utility.Mock;
+    using System.Collections.Generic;
+
+    public class ReverseCollisionsOrderTest
+    {
+        private GameObject containingObject;
+        private ReverseCollisionsOrder subject;
+
+        [SetUp]
+        public void SetUp()
+        {
+            containingObject = new GameObject();
+            subject = containingObject.AddComponent<ReverseCollisionsOrder>();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            Object.DestroyImmediate(subject);
+            Object.DestroyImmediate(containingObject);
+        }
+
+        [Test]
+        public void Reverse()
+        {
+            UnityEventListenerMock reversedMock = new UnityEventListenerMock();
+            subject.Reversed.AddListener(reversedMock.Listen);
+
+            List<CollisionNotifier.EventData> collisionList = new List<CollisionNotifier.EventData>();
+            GameObject oneContainer;
+            CollisionNotifier.EventData oneData = GenerateCollision(out oneContainer);
+            oneContainer.name = "one";
+            collisionList.Add(oneData);
+
+            GameObject twoContainer;
+            CollisionNotifier.EventData twoData = GenerateCollision(out twoContainer);
+            twoContainer.name = "two";
+            collisionList.Add(twoData);
+
+            GameObject threeContainer;
+            CollisionNotifier.EventData threeData = GenerateCollision(out threeContainer);
+            threeContainer.name = "three";
+            collisionList.Add(threeData);
+
+            ActiveCollisions.EventData eventData = new ActiveCollisions.EventData().Set(collisionList);
+
+            Assert.AreEqual("one,two,three,", GetListNames(eventData));
+            Assert.IsFalse(reversedMock.Received);
+
+            ActiveCollisions.EventData reversedList = subject.Reverse(eventData);
+
+            Assert.IsTrue(reversedMock.Received);
+            Assert.AreEqual("three,two,one,", GetListNames(reversedList));
+
+            Object.DestroyImmediate(oneContainer);
+            Object.DestroyImmediate(twoContainer);
+            Object.DestroyImmediate(threeContainer);
+        }
+
+        protected string GetListNames(ActiveCollisions.EventData list)
+        {
+            string returnString = "";
+            foreach (CollisionNotifier.EventData data in list.activeCollisions)
+            {
+                returnString += data.collider.name + ",";
+            }
+            return returnString;
+        }
+
+        protected CollisionNotifier.EventData GenerateCollision(out GameObject container)
+        {
+            container = new GameObject();
+            BoxCollider boxCollider = container.AddComponent<BoxCollider>();
+            Collider collider = container.GetComponent<Collider>();

```

Comment:
See my comment above.

Additionally: This is duplicated code. Perhaps this method should live in a shared class in the `Test` namespace instead?